### PR TITLE
Pvt model

### LINKED ISSUE #17596: Pyramid Vision Transformer
### Model description

I would like to add the[ Pyramid Vision Transformer model](https://arxiv.org/abs/2102.12122).

#### Paper Abstract
Pyramid Vision Transformer~(PVT), has several merits compared to prior arts. (1) Different from ViT that typically has low-resolution outputs and high computational and memory cost, PVT can be not only trained on dense partitions of the image to achieve high output resolution, which is important for dense predictions but also using a progressive shrinking pyramid to reduce computations of large feature maps. (2) PVT inherits the advantages from both CNN and Transformer, making it a unified backbone in various vision tasks without convolutions by simply replacing CNN backbones. (3) We validate PVT by conducting extensive experiments, showing that it boosts the performance of many downstream tasks

### Open source status

- [X] The model implementation is available
- [x] The model weights are available

### Provide useful links for the implementation

Model Implementation: https://github.com/whai362/PVT

Pretrained model weights for semantic segmentation: https://github.com/whai362/PVT/tree/v2/segmentation (based on ADE20K)

### PR DESCRIPTION
**Pull request by Xrenya**
# Add PVT(Pyramid Vision Transformer)

Partially fixes: [issue](https://github.com/huggingface/transformers/issues/17596), [Closed PR](https://github.com/huggingface/transformers/pull/22445)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

@amyeroberts

From previous PR:

> PvtBlock which contains a PvtPatchEmbeddings layer and the subsequent PvtLayer layers. 
PvtLayer has depth, while PvtPatchEmbeddings only at the beginning of the each encoder block and it would be trailed along the whole depth without using or it would require extra logic to make it None and also would require to trail height and width.

### CODE REVIEW

**Review by amyeroberts — APPROVED** [review]
Thanks for adding this model! 

Just a few small nits to tidy up. After that, I think we're good to merge!

File: docs/source/en/index.md [hunk 1]
```diff
@@ -194,6 +194,7 @@ The documentation is organized into five sections:
 1. **[PLBart](model_doc/plbart)** (from UCLA NLP) released with the paper [Unified Pre-training for Program Understanding and Generation](https://arxiv.org/abs/2103.06333) by Wasi Uddin Ahmad, Saikat Chakraborty, Baishakhi Ray, Kai-Wei Chang.
 1. **[PoolFormer](model_doc/poolformer)** (from Sea AI Labs) released with the paper [MetaFormer is Actually What You Need for Vision](https://arxiv.org/abs/2111.11418) by Yu, Weihao and Luo, Mi and Zhou, Pan and Si, Chenyang and Zhou, Yichen and Wang, Xinchao and Feng, Jiashi and Yan, Shuicheng.
 1. **[ProphetNet](model_doc/prophetnet)** (from Microsoft Research) released with the paper [ProphetNet: Predicting Future N-gram for Sequence-to-Sequence Pre-training](https://arxiv.org/abs/2001.04063) by Yu Yan, Weizhen Qi, Yeyun Gong, Dayiheng Liu, Nan Duan, Jiusheng Chen, Ruofei Zhang and Ming Zhou.
+1. **[Pvt](model_doc/pvt)** (from Nanjing University, The University of Hong Kong etc.) released with the paper [Pyramid Vision Transformer: A Versatile Backbone for Dense Prediction without Convolutions](https://arxiv.org/pdf/2102.12122.pdf) by Wenhai Wang, Enze Xie, Xiang Li, Deng-Ping Fan, Kaitao Song, Ding Liang, Tong Lu, Ping Luo, Ling Shao..
```
[the remark this answers is not in the document]
Comment (Xrenya) [reply docs/source/en/index.md]: Done!

File: docs/source/en/model_doc/pvt.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+object detection, instance and semantic segmentation. For example, with a comparable number of parameters, PVT+RetinaNet 
+achieves 40.4 AP on the COCO dataset, surpassing ResNet50+RetinNet (36.3 AP) by 4.1 absolute AP (see Figure 2). We hope 
+that PVT could serve as an alternative and useful backbone for pixel-level predictions and facilitate future research.*
+
+This model was contributed by [Xrenya](<https://huggingface.co/Xrenya). The original code can be found [here](https://github.com/whai362/PVT).
+
+Tips:
+
+- The PyTorch version of this model is only available.
+
```
[the remark this answers is not in the document]
Comment (Xrenya) [reply docs/source/en/model_doc/pvt.md]: Done!

File: src/transformers/models/auto/configuration_auto.py [hunk 3]
```diff
@@ -555,6 +557,7 @@
         ("plbart", "PLBart"),
         ("poolformer", "PoolFormer"),
         ("prophetnet", "ProphetNet"),
+        ("pvt", "Pvt"),
```
[the remark this answers is not in the document]
Comment (Xrenya) [reply src/transformers/models/auto/configuration_auto.py]: Done!

File: tests/models/pvt/test_modeling_pvt.py [hunk 4]
```diff
@@ -0,0 +1,496 @@
...
+
+    def test_for_image_classification(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_for_image_classification(*config_and_inputs)
+
+    @slow
+    def test_model_from_pretrained(self):
+        for model_name in PVT_PRETRAINED_MODEL_ARCHIVE_LIST[:1]:
+            model = PvtForImageClassification.from_pretrained(model_name)
+            self.assertIsNotNone(model)
```
[the remark this answers is not in the document]
Comment (Xrenya) [reply tests/models/pvt/test_modeling_pvt.py]: Done!

File: src/transformers/models/pvt/convert_pvt_to_pytorch.py [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+    # load HuggingFace model
+    model = PvtForImageClassification(config).eval()
+    model.load_state_dict(state_dict)
+
+    # Check outputs on an image, prepared by PVTFeatureExtractor
+    feature_extractor = PvtImageProcessor(size=config.image_size)
+    encoding = feature_extractor(images=prepare_img(), return_tensors="pt")
+    pixel_values = encoding["pixel_values"]
+    outputs = model(pixel_values)
+    outputs = outputs.logits.detach().cpu().argmax().item()
```
[the remark this answers is not in the document]
Comment (Xrenya) [reply src/transformers/models/pvt/convert_pvt_to_pytorch.py]: Done!

File: src/transformers/models/pvt/convert_pvt_to_pytorch.py [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+
+    Path(pytorch_dump_folder_path).mkdir(exist_ok=True)
+    print(f"Saving model pytorch_model.bin to {pytorch_dump_folder_path}")
+    model.save_pretrained(pytorch_dump_folder_path)
+    print(f"Saving feature extractor to {pytorch_dump_folder_path}")
+    feature_extractor.save_pretrained(pytorch_dump_folder_path)
```
[the remark this answers is not in the document]
Comment (Xrenya) [reply src/transformers/models/pvt/convert_pvt_to_pytorch.py]: Done!

File: tests/models/pvt/test_modeling_pvt.py [hunk 7]
```diff
@@ -0,0 +1,496 @@
...
+    def test_inputs_embeds(self):
+        pass
+
+    @unittest.skip("Pvt does not have get_input_embeddings method and get_output_embeddings methods")
+    def test_model_common_attributes(self):
+        pass
+
+    @unittest.skip("PVT normally initialized the parameters so negative values are expected.")
+    def test_initialization(self):
+        pass
```
[the remark this answers is not in the document]
Comment (Xrenya) [reply tests/models/pvt/test_modeling_pvt.py]: Done!

**Review by Xrenya — COMMENTED** [review]
I think I have changed everything was requested

File: docs/source/en/model_doc/pvt.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+specific language governing permissions and limitations under the License.
+-->
+
+# Pyramid Vision Transformer (PVT)
+
+## Overview
+
+The PVT model was proposed in
+[Pyramid Vision Transformer: A Versatile Backbone for Dense Prediction without Convolutions](https://arxiv.org/abs/2102.12122)
+by Wenhai Wang, Enze Xie, Xiang Li, Deng-Ping Fan, Kaitao Song, Ding Liang, Tong Lu, Ping Luo, Ling Shao.
```
Comment (amyeroberts) [inline docs/source/en/model_doc/pvt.md:19]: We should have a few sentences here explaining clearly what the model does and how it does it. 

A good example is Music Gen: https://huggingface.co/docs/transformers/main/en/model_doc/musicgen

**Review by amyeroberts — APPROVED** [review]
Thanks for all your work adding this model! 

Just one last thing to add - a description for the model in the README and then we're good to merge 🤗

**Comment by amyeroberts** [comment]
@Xrenya Thanks again for adding and iterating. Merging now :)

**Pull request merged**